test/functional/test-viewer.js

```diff
@@ -226,6 +230,100 @@ describe('Viewer', () => {
     expect(viewportEvent).to.not.equal(null);
   });
 
+  describe('replaceUrl', () => {
+    function setUrl(href) {
+      const url = parseUrl(href);
+      windowApi.location.href = url.href;
+      windowApi.location.hash = url.hash;
+    }
+
+    it('should replace URL for the same non-proxy origin', () => {
+      const fragment = '#replaceUrl=http://www.example.com/two&b=1';
+      setUrl('http://www.example.com/one' + fragment);
+      new Viewer(ampdoc);
+      expect(windowApi.history.replaceState).to.be.calledOnce;
+      expect(windowApi.history.replaceState).to.be.calledWith({}, '',
+          'http://www.example.com/two' + fragment);
+      expect(ampdoc.getUrl())
+          .to.equal('http://www.example.com/two' + fragment);
```

Comment:
got it. Might be worth adding a test for `fragment = '#replaceUrl=http://www.example.com/two%25b=2&b=1';` to expect `http://www.example.com/two&b=2` then.